### PR TITLE
Resolve urls

### PR DESCRIPTION
**Pull request by ellesmith88**
Changes to allow DatasetMapper to handle URLs as well as glob pattern such as `"/badc/cmip6/data/.../foo/baa/{full_file_name_1.nc;full_file_name_2.nc}"`

### CODE REVIEW

**Comment by ellesmith88** [comment]
@agstephens I've updated this now - one thing I wasn't too sure how to do was how to adjust `BaseParameter` / `CollectionParameter` to allow an instance of `FileMapper` to pass through. 

The problem came in `daops`:  https://github.com/roocs/daops/blob/6f54de688a06cc4272b8cab82bcf9e16ce9e8205/daops/utils/consolidate.py#L28-L33
as `consolidate` iterates through the collection. I've made the result of collection a list when a `FileMapper` object is passed through.

see here:
https://github.com/roocs/roocs-utils/blob/9b69aa41e267f0eb13dac0bbdebca03fa96bb21c/roocs_utils/parameter/collection_parameter.py#L27-L29

but thought you might have some thoughts on how to improve this?

**Comment by ellesmith88** [comment]
@agstephens I've run the current `master` of `clisops` with this branch (after updating the parameter classes) and all the tests pass - are you happy to merge once the travis/docs pass?

File: roocs_utils/project_utils.py [hunk 1]
```diff
@@ -18,7 +19,8 @@ def __init__(self, dset, project=None, force=False):
         | dset must be a string and can be input as:
         | A dataset ID: e.g. "cmip5.output1.INM.inmcm4.rcp45.mon.ocean.Omon.r1i1p1.latest.zostoga"
         | A file path: e.g. "/badc/cmip5/data/cmip5/output1/MOHC/HadGEM2-ES/rcp85/mon/atmos/Amon/r1i1p1/latest/tas/tas_Amon_HadGEM2-ES_rcp85_r1i1p1_200512-203011.nc"
-        | A path to a group of files: e.g. "/badc/cmip5/data/cmip5/output1/MOHC/HadGEM2-ES/rcp85/mon/atmos/Amon/r1i1p1/latest/tas/*.nc" or "/badc/cmip5/data/cmip5/output1/MOHC/HadGEM2-ES/rcp85/mon/atmos/Amon/r1i1p1/latest/tas"
+        | A path to a group of files: e.g. "/badc/cmip5/data/cmip5/output1/MOHC/HadGEM2-ES/rcp85/mon/atmos/Amon/r1i1p1/latest/tas/*.nc" or "/badc/cmip6/data/CMIP6/CMIP/MIROC/MIROC6/amip/r1i1p1f1/day/tas/gn/latest/{tas_day_MIROC6_amip_r1i1p1f1_gn_19790101-19881231.nc;tas_day_MIROC6_amip_r1i1p1f1_gn_19890101-19981231.nc}"
+        | A directory e.g. "/badc/cmip5/data/cmip5/output1/MOHC/HadGEM2-ES/rcp85/mon/atmos/Amon/r1i1p1/latest/tas"
```
Comment (agstephens) [inline roocs_utils/project_utils.py:23]: @ellesmith88 Please add in here that the input can be an instance of FileMapper.

```
      | An instance of the FileMapper class (that represents a set of files within a single directory)
```

**Comment by agstephens** [comment]
@ellesmith88 it looks good. Just one comment, to add a comment to the DatasetMapper docstring:

```

      | An instance of the FileMapper class (that represents a set of files within a single directory)
```

And then we are good to go. Thanks

**Pull request merged**